python/cudf/cudf/core/index.py

```diff
@@ -1641,10 +1832,106 @@ def as_index(arbitrary, **kwargs):
     )
 
 
-def _setdefault_name(values, kwargs):
-    if "name" not in kwargs:
+def _setdefault_name(values, **kwargs):
+    if "name" not in kwargs or kwargs["name"] is None:
         if not hasattr(values, "name"):
-            kwargs.setdefault("name", None)
+            kwargs.update({"name": None})
         else:
-            kwargs.setdefault("name", values.name)
+            kwargs.update({"name": values.name})
     return kwargs
+
+
+def Int64Index(data=None, dtype="int64", copy=False, name=None):
+    """Immutable ndarray implementing an ordered, sliceable set.
```

Comment:
The Pandas description of `Int64Index` is definitely weird:

1. Indexes are not ndarrays (I guess MultiIndexes are kind of like ndarrays, but still not really)
2. Indexes are not "sets" in that they are not `set`-like

So maybe something like "Immutable, ordered and sliceable sequence of integer labels"?